### PR TITLE
update Batch Changes and Code Insights getting-started pages

### PR DESCRIPTION
**Pull request by sqs**
* Previously, these CTAs said you needed to install Sourcegraph self-hosted to use Batch Changes or Code Insights. Now they mention Cloud first.
* Removes the hardcoded "requirements" that become out of date and aren't useful.
* Simplifies the Batch Changes and Code Insights getting-started pages. (See commit messages for more details.)

# CTAs

## Before

image
image

## After

image
image

# Simpler Batch Changes getting started page

(This screenshot is from dotcom. For self-hosted instances, the bottom CTA is different.)

image

# Simpler Code Insights getting started page

(This screenshot is from dotcom. For self-hosted instances, the left-hand figure is interactive and is not a prerecorded video.)

image

---

## Test plan

View `/batch-changes` and `/insights` in dotcom mode.

## App preview:

- [Web](https://sg-web-sqs-install-cta.onrender.com/search)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.

### CODE REVIEW

**Comment by sg-e2e-regression-test-bob** [comment]
## Bundle size report 📦

| Initial size | Total size | Async size | Modules |
| --- | --- | --- | --- |
| 0.00% (0.00 kb) | 0.00% (+0.53 kb) | 0.00% (+0.53 kb) | 0.00% (0) |

Look at the [Statoscope report](https://console.cloud.google.com/storage/browser/_details/sourcegraph_reports/statoscope-reports/sqs/install-cta/compare-report.html;tab=live_object) for a full comparison between the commits 0830345 and e659285 or [learn more](https://docs.sourcegraph.com/dev/how-to/testing#bundlesize).


<details>
<summary>Open explanation</summary>

- `Initial size` is the size of the initial bundle (the one that is loaded when you open the page)
- `Total size` is the size of the initial bundle + all the async loaded chunks
- `Async size` is the size of all the async loaded chunks
- `Modules` is the number of modules in the initial bundle
</details>

**Comment by sqs** [comment]
@Joelkw @felixfbecker I see some `licensed` checks that determine whether to show a `+ Create your first insight` button, and whether to show `Explore template` or `Use template`. All instances get a free tier of 2 global insights, so shouldn't we just always show that button and `Use template`, and let the product inform the user if they have exceeded the licensed limits?

**Comment by sqs** [comment]
Reviewers: because this is code that affects marketing CTAs and not functionality, you can do a less-intensive review.

**Comment by Joelkw** [comment]
@sqs to answer the specific CTA questions – 

>  All instances get a free tier of 2 global insights, so shouldn't we just always show that button and Use template, and let the product inform the user if they have exceeded the licensed limits?

Fine with these changes! 

(Details for anyone else: the product does note on the creation form if you've hit your limit as well. There was some thought on not leading users on (they can't use the template, because they can't save it) but that's small and we likely over-corrected in that direction given beta-to-GA fears of pricing pushback as we exited free beta. For the main CTA, the + Create first insight is replaced with a "schedule demo" if you don't have a license. We've never had anyone actually schedule it, however, so for now it's fine to revert this as well, at least until we someday focus on in-product conversions when we're further along the PLG journey.)

**Review by Joelkw — APPROVED** [review]

File: client/web/src/enterprise/batches/global/GlobalBatchChangesArea.tsx [hunk 1]
```diff
@@ -47,12 +45,9 @@ const BatchChangeClosePage = lazyComponent<BatchChangeClosePageProps, 'BatchChan
     () => import('../close/BatchChangeClosePage'),
     'BatchChangeClosePage'
 )
-const DotcomGettingStartedPage = lazyComponent<DotcomGettingStartedPageProps, 'DotcomGettingStartedPage'>(
```
Comment (eseliger) [inline client/web/src/enterprise/batches/global/GlobalBatchChangesArea.tsx:50]: I think we want to keep this special-case page for dotcom. The content of getting started can be 100% the same, but it serves the following purposes:

- It doesn't register additional routes
- It is a version that doesn't show the list page as an option

After this change, `https://sourcegraph.com/batch-changes/create` will be available (but not functional).
Also, the getting started page looks broken:
Screenshot 2022-11-08 at 11 40 38@2x

**Review by eseliger — CHANGES_REQUESTED** [review]
Content changes LGTM, but I think we need to revert the removal of the DotcomGettingStartedPage, see comment inline

File: client/web/src/enterprise/batches/global/GlobalBatchChangesArea.tsx (hunk 1 above)
Comment (eseliger) [reply client/web/src/enterprise/batches/global/GlobalBatchChangesArea.tsx:50]: if you want one of batch changes to take this over, lmk.

**Review by vovakulikov — COMMENTED** [review]
I'm ok with code insights changes from tech point of view but I would like to get a quick check from @AlicjaSuska about design changes on Code Insights landing pages. 

A few points from me about this PR.

- I see that you removed introduction videos about code insights as well, I haven't found there any outdated information about cloud or install requirements though. Personally I've found this video very useful. 
- Template section is also might be useful in order to give a quick sense of what code insights capable of 
- Footer links, I think it was one of most noticeable places where we had link to our documentation, do we really want to remove them? (find with removing from landing page but in this case I would add theme somewhere else) 

cc @AlicjaSuska

File: client/web/src/enterprise/batches/global/GlobalBatchChangesArea.tsx (hunk 1 above)
Comment (sqs) [reply client/web/src/enterprise/batches/global/GlobalBatchChangesArea.tsx:50]: @eseliger I want to remove disparities between dotcom and non-dotcom as much as possible, even if it means a bit of jankiness like that error message. I think our audience will understand that, and it paves the way for things like read-only access or demos for some of the UI (not in scope now, but possibly something we'd like in the future to work on on Growth Marketing). I will update that error message to say that you need to use Sourcegraph Cloud or self-hosted, and then I think it's fine.

**Comment by sqs** [comment]
@vovakulikov:

* I will link the videos from the docs (should have added that in this PR, will add it now). The page was very long IMO and we want to be driving people to our docs, which always have the latest videos and other materials.
* Code Insights templates still show up on self-hosted instance (just not on dotcom anymore, but users can still go to the docs).
* The docs are now linked even more prominently IMO, above the fold under `Resources`.

File: client/web/src/enterprise/batches/global/GlobalBatchChangesArea.tsx (hunk 1 above)
Comment (sqs) [reply client/web/src/enterprise/batches/global/GlobalBatchChangesArea.tsx:50]: I also just made it so the `/create` route is not registered on dotcom (with just an if-statement in the JSX).

**Comment by sqs** [comment]
@AlicjaSuska I'm going to merge this because 98% of the changes should be good and the rest are marketing copy/layout-related, so easy to change if you have other ideas.

**Pull request merged**